lang/java/avro/src/main/java/org/apache/avro/SchemaNormalization.java

```diff
@@ -17,31 +17,60 @@
  */
 package org.apache.avro;
 
+import org.apache.avro.util.internal.JacksonUtils;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashSet;
 import java.util.Map;
 import java.util.HashMap;
 import java.io.IOException;
 import java.io.UnsupportedEncodingException;
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
 
-/** Collection of static methods for generating the canonical form of
- * schemas (see {@link #toParsingForm}) -- and fingerprints of canonical
- * forms ({@link #fingerprint}).
+/** Collection of static methods for generating the parser canonical form of
+ * schemas (see {@link #toParsingForm}), standard canonical form of schemas
+ * (see {@link #toCanonicalForm(Schema)} or {@link #toCanonicalForm(Schema, LinkedHashSet)})
+ * with user defined Logical Types (see {@link #setLogicalTypes(LogicalTypes...)} and
+ * fingerprints of canonical forms ({@link #fingerprint}).
  */
 public class SchemaNormalization {
 
   private SchemaNormalization() {}
 
+  private static final LinkedHashSet<String> RESERVED_PROPERTIES = new LinkedHashSet<>();
+
+  private static final LinkedHashSet<LogicalTypes> ADDITIONAL_LOGICAL_TYPES = new LinkedHashSet<>();
+
+  // Avro standard reserved properties
+  static {
+    Collections.addAll(RESERVED_PROPERTIES,
+      "name", "type", "fields", "symbols", "items", "values",
+      "logicalType", "size", "order", "doc", "aliases", "default");
+  }
+
+  // Set user defined logical types for canonical form of schema definition.
```

Comment:
We should avoid mutating static state like this.  If this is to vary from application to application, then it should probably either be an instance variable or a parameter of toCanonicalForm(), not a global, static variable.

Reply:
Updated the `Spec` for both canonical form transformation.